examples/seq2seq/finetune_trainer.py

```diff
@@ -0,0 +1,423 @@
+        trim_batch,
+        use_task_specific_params,
+    )
+
+
+logger = logging.getLogger(__name__)
+
+
+class Seq2SeqDataCollator:
+    def __init__(self, tokenizer, data_args):
+        self.tokenizer = tokenizer
+        self.pad_token_id = tokenizer.pad_token_id
+        self.data_args = data_args
+        self.add_prefix_space = isinstance(tokenizer, BartTokenizer)
+
+    def __call__(self, batch) -> Dict[str, torch.Tensor]:
+        if hasattr(self.tokenizer, "prepare_seq2seq_batch"):
+            batch = self._encode(batch)
+            input_ids, attention_mask, target_ids = (
+                batch["input_ids"],
+                batch["attention_mask"],
+                batch["decoder_input_ids"],
+            )
+        else:
+            input_ids = torch.stack([x["input_ids"] for x in batch])
+            attention_mask = torch.stack([x["attention_mask"] for x in batch])
+            target_ids = torch.stack([x["decoder_input_ids"] for x in batch])
+
+            target_ids = trim_batch(target_ids, self.pad_token_id)
+            input_ids, attention_mask = trim_batch(input_ids, self.pad_token_id, attention_mask=attention_mask)
+
+        if isinstance(self.tokenizer, T5Tokenizer):
+            decoder_input_ids = self._shift_right_t5(target_ids)
+            labels = target_ids
+        else:
+            decoder_input_ids = shift_tokens_right(target_ids, self.pad_token_id)
+            labels = target_ids
+
+        batch = {
+            "input_ids": input_ids,
+            "attention_mask": attention_mask,
+            "decoder_input_ids": decoder_input_ids,
+            "labels": labels,
+        }
+        return batch
+
+    def _shift_right_t5(self, input_ids):
+        decoder_start_token_id = self.pad_token_id
+
+        assert (
+            decoder_start_token_id is not None
+        ), "self.model.config.decoder_start_token_id has to be defined. In T5 it is usually set to the pad_token_id. See T5 docs for more information"
+
+        # shift inputs to the right
+        shifted_input_ids = input_ids.new_zeros(input_ids.shape)
+        shifted_input_ids[..., 1:] = input_ids[..., :-1].clone()
+        shifted_input_ids[..., 0] = decoder_start_token_id
+
+        return shifted_input_ids
+
+    def _encode(self, batch) -> Dict[str, torch.Tensor]:
+        batch_encoding = self.tokenizer.prepare_seq2seq_batch(
+            [x["src_texts"] for x in batch],
+            src_lang=self.data_args.src_lang,
+            tgt_texts=[x["tgt_texts"] for x in batch],
+            tgt_lang=self.data_args.tgt_lang,
+            max_length=self.data_args.max_source_length,
+            max_target_length=self.data_args.max_target_length,
+            return_tensors="pt",
+            add_prefix_space=self.add_prefix_space,
+        )
+        return batch_encoding.data
+
+
+@dataclass
+class Seq2SeqTrainingArguments(TrainingArguments):
+    label_smoothing: Optional[float] = field(default=0.0, metadata={"help": "Label smoothing"})
```

Comment:
Documentation  of those three args would be good (or expand the help comments a bit more). For instance, is `label_smoothing` the epsilon for label smoothing?

Reply:
Yes , `label_smoothing` is the epsilon. doc and help comments need to be refined. Focusing on getting the functionality right first. Will definitely ping you for doc reviews :)